src/run.rs

```diff
@@ -314,6 +324,7 @@ mod tests {
             rate_limit_per_second: None,
             operation_factory: Arc::new(FnOperationFactory(f)),
             max_retries_per_op: 0,
+            max_retries_in_total: usize::MAX,
```

Comment:
By default `retiries_in_total` are unlimited, but `retries_per_op` are set to 0. This is a bit inconsistent.

Reply:
I didn't use `max_retries_in_total` here because I didn't want to change the existing tests that used `max_retries_per_op`, however I don't think it's very bad so ultimately I changed `max_retries_in_total` here to 0 and adjusted the affected tests.